src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestValidationParameters.cs

```diff
@@ -80,6 +80,16 @@ namespace Microsoft.IdentityModel.Protocols.SignedHttpRequest
     /// <returns>Expected to throw an appropriate exception if SignedHttpRequest replay is detected.</returns>
     public delegate Task ReplayValidatorAsync(SecurityToken signedHttpRequest, SignedHttpRequestValidationContext signedHttpRequestValidationContext, CancellationToken cancellationToken);
 
+    /// <summary>
+    /// A delegate that will take control over SignedHttpRequest nonce validation, if set.
+    /// </summary>
+    /// <param name="key">the key use to validate server nonce.</param>
+    /// <param name="signedHttpRequest">A SignedHttpRequest.</param>
+    /// <param name="signedHttpRequestValidationContext">A structure that wraps parameters needed for SignedHttpRequest validation.</param>
+    /// <param name="cancellationToken">Propagates notification that operations should be canceled.</param>
+    /// <returns>Expected to throw an appropriate exception if SignedHttpRequest replay is detected.</returns>
+    public delegate bool NonceValidatorAsync(SecurityKey key, SecurityToken signedHttpRequest, SignedHttpRequestValidationContext signedHttpRequestValidationContext, CancellationToken cancellationToken);
```

Comment:
IIRC, Brent said it share the same parameter with Popkeyresolver, not sure the first parameter should be security key or security token.

Reply:
Good for @brentschmaltz to clarify, I think in some respects we could just have both available.

Reply:
This should also take in the nonce claim string

Reply:
SecurityKey was correct.